lib/temporal/scheduled_thread_pool.rb

```diff
@@ -0,0 +1,111 @@
+require 'temporal/metric_keys'
+
+# This class implements a thread pool for scheduling tasks with a delay.
+# If threads are all occupied when a task is scheduled, it will be queued
+# with the sleep delay adjusted based on the wait time. 
+module Temporal
+  class ScheduledThreadPool
```

Comment:
@jeffschoner is this something you all have implemented internally and tested in other services?

Reply:
@DeRauk This has not run yet in production services yet. I have done testing in our staging environment. CI for all our tests pass as well. Instead of the `LocalWorkflowContext`-based unit testing mode, almost all of our workflow tests in CI use real workers against the Temporal emulator from the Java SDK. Our CI does also look for problems like thread leaks.

My plan is to deploy this next week with a max heartbeat throttle timeout of 0, which will use the new code path but effectively disable throttling. This setting is behind a feature flag that we'll use to roll out the 60s default max to all namespaces gradually. I'm glad to hold off on merging this until that process is done to have a higher level of confidence in the change.